test/inductor/test_triton_kernels.py

```diff
@@ -3550,6 +3550,279 @@ def f(x, increment):
+
+        def perf_model(*args, **kwargs):
+            records["run_perf_model"] = True
+            return kwargs["BLOCK_SIZE"] * -1
+
+        if with_perf_model:
+            prune_configs_by = {"perf_model": perf_model, "top_k": 1}
+        else:
+            prune_configs_by = {"early_config_prune": early_config_prune}
+
+        @triton.autotune(
+            configs=[
+                triton.Config(kwargs={"BLOCK_SIZE": 32}),
+                triton.Config(kwargs={"BLOCK_SIZE": 128}),
+            ],
+            key=["N"],
+            prune_configs_by=prune_configs_by,
+        )
+        @triton.jit
+        def prune_by_kernel(
+            dst,
+            src,
+            add_float,
+            string_arg: tl.constexpr,
+            bool_var: tl.constexpr,
+            N,
+            BLOCK_SIZE: tl.constexpr,
+        ):
+            offsets = tl.program_id(0) * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+            x = tl.load(src + offsets, mask=offsets < N)
+            # Let's make sure we always select a block size of 128 based on our perf_model
+            if (string_arg == "TEST" and bool_var) and BLOCK_SIZE == 128:
+                x = x + add_float
+            tl.store(dst + offsets, x, mask=offsets < N)
+
+        torch._dynamo.reset()
+        counter = 0
+
+        # Track how many compiles occur
+        @torch._dynamo.on_compile_start
+        def start_callback():
+            nonlocal counter
+            counter += 1
+
+        @torch.compile(fullgraph=True, backend=backend)
+        def f(dst, src, add_float, string_arg, bool_var, N):
+            grid = lambda META: (triton.cdiv(N, META["BLOCK_SIZE"]),)
+            prune_by_kernel[grid](dst, src, add_float, string_arg, bool_var, N=N)
+
+        N = 1024
+        src = torch.randn(N, device=GPU_TYPE)
+        dst = torch.empty(N, device=GPU_TYPE)
+
+        # first compilation, this prunes the configs
+        f(dst, src, 1.5, "TEST", True, N)
+
+        self.assertEqual(counter, 1)
+
+        # configs has changed (because of the pruning), so this will recompile
+        f(dst, src, 1.5, "TEST", True, N)
+
+        self.assertEqual(counter, 2)
+
+        # this should not trigger a recompile
+        # as the configs have not changed here
+        f(dst, src, 1.5, "TEST", True, N)
+
+        self.assertEqual(counter, 2)
+
+        # Modify the autotuner object
+        prune_by_kernel.configs = [triton.Config(kwargs={"BLOCK_SIZE": 64})]
+
+        # Calling the kernel after modifying the autotuner should
+        # trigger a recompile
+        f(dst, src, 1.5, "TEST", True, N)
+
+        # We should observe three compile events if guards are installed correctly
+        self.assertEqual(counter, 3)
+
+    # Test if non-strict export works
+    @requires_gpu
+    @common_utils.parametrize("with_perf_model", [True, False])
+    def test_triton_prune_configs_non_strict_export(self, with_perf_model):
```

Comment:
Can we merge this test with the first one? Seems that the only difference is the `@torch.library.triton_op` decorator, which we can apply on a boolean flag?